Mathlib/CategoryTheory/FiberedCategory/HomLift.lean

```diff
@@ -0,0 +1,244 @@
+/-
+Copyright (c) 2024 Calle Sönne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Lezeau, Calle Sönne
+-/
+
+import Mathlib.CategoryTheory.Functor.Category
+import Mathlib.CategoryTheory.CommSq
+
+/-!
+
+# HomLift
+
+Given a functor `p : 𝒳 ⥤ 𝒮`, this file provides API for expressing the fact that `p(φ) = f`
+for given morphisms `φ` and `f`.
+
+## Main definition
+
+Given morphism `φ : a ⟶ b` in `𝒳` and `f : R ⟶ S` in `𝒮`, `p.IsHomLift f φ` is defined as a
+structure containing the data that `p(a) = R`, `p(b) = S` and the fact that the following square
+commutes
+```
+  p(a) --p(φ)--> p(b)
+  |               |
+  |               |
+  v               v
+  R -----f------> S
+```
+where the vertical arrows are given by `eqToHom` corresponding to the equalities `p(a) = R` and
+`p(b) = S`.
+
+-/
+
+universe u₁ v₁ u₂ v₂
+
+open CategoryTheory Category
+
+variable {𝒮 : Type u₁} {𝒳 : Type u₂} [Category 𝒳] [Category 𝒮]
+
+namespace CategoryTheory
+
+/-- The proposition that an arrow a --φ--> b lifts an arrow R --f--> S in 𝒮 via p. This is
```

Comment:
The docstring should make more explicit in which category `φ` is and what are the source/target categories for `p`.